### PR TITLE
test: add unit tests for /get_mcq endpoint edge cases

### PR DESCRIPTION
**Pull request by toyes3429**
###  Addressed Issues:
Provides test coverage for #463

###  Screenshots/Recordings:
backend-test-cases

###  Additional Notes:
This PR adds backend unit tests that document the current behavior of the /get_mcq endpoint when fewer or zero MCQs are generated than requested.

These tests do not modify API responses or generation logic, but provide a foundation for safely implementing the metadata exposure proposed in #463.

## Checklist

- [x] My PR addresses a single issue, fixes a single bug or makes a single improvement.
- [x] My code follows the project's code style and conventions
- [ ] If applicable, I have made corresponding changes or additions to the documentation
- [x] If applicable, I have made corresponding changes or additions to tests
- [x] My changes generate no new warnings or errors
- [x] I have joined the [Discord server](https://discord.gg/hjUhu33uAn) and I will share a link to this PR with the project maintainers there
- [x] I have read the [Contribution Guidelines](../CONTRIBUTING.md)
- [x] Once I submit my PR, CodeRabbit AI will automatically review it and I will address CodeRabbit's comments.

## AI Usage Disclosure

Check one of the checkboxes below:

- [ ] This PR does not contain AI-generated code at all.
- [x] This PR contains AI-generated code. I have tested the code locally and I am responsible for it.

I have used the following AI models and tools: TODO
ChatGPT (for test structure guidance and edge-case reasoning)

### Testing
- pytest backend/tests
- Tests run without loading ML models
- Verified /get_mcq partial generation behavior

## Summary by CodeRabbit

* **Tests**
  * Added test configuration that stubs heavy generator components before importing the app, enabling isolated, deterministic backend tests and faster CI feedback.
  * Added a unit test for the MCQ generation endpoint to validate partial-generation behavior, response structure, and HTTP status.

### CODE REVIEW

File: backend/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+sys.modules["Generator"] = fake_generator
+sys.modules["Generator.main"] = fake_main
+sys.modules["Generator.question_filters"] = fake_filters
+
+# -------------------------------------------------
+# Safe import
+# -------------------------------------------------
+
+from backend.server import app
```
[the remark this answers is not in the document]
Comment (toyes3429) [reply backend/tests/conftest.py:99]: i have already added the __init__.py file
